lib/daru/io/exporters/excel.rb

```diff
@@ -23,34 +41,95 @@ class Excel < Base
+        #   df = Daru::DataFrame.new([[1,2],[3,4]], order: [:a, :b])
+        #
+        #   #=> #<Daru::DataFrame(2x2)>
+        #   #      a   b
+        #   #  0   1   3
+        #   #  1   2   4
+        #
+        #   Daru::IO::Exporters::Excel.new(df,
+        #     "dataframe_df.xls",
+        #     header: { color: :red, weight: :bold },
+        #     index:  false,
+        #     data:   { color: :blue }
+        #   ).call
+        #
+        # @example Writing a DataFrame with Multi-Index to an Excel file
+        #   df = Daru::DataFrame.new [[1,2],[3,4]], order: [:x, :y], index: [[:a, :b, :c], [:d, :e, :f]]
+        #
+        #   #=> #<Daru::DataFrame(2x2)>
+        #   #             x   y
+        #   #  a  b   c   1   3
+        #   #  d  e   f   2   4
+        #
+        #   Daru::IO::Exporters::Excel.new(df,
+        #     "dataframe_df.xls",
+        #     header: { color: :red, weight: :bold },
+        #     index:  { color: :green },
+        #     data:   { color: :blue }
+        #   ).call
+        def initialize(dataframe, path, header: true, data: true, index: true)
           optional_gem 'spreadsheet', '~> 1.1.1'
 
           super(dataframe)
-          @path      = path
-          @options   = options
+          @path   = path
+          @data   = data
+          @index  = index
+          @header = header
         end
 
-        # @note
-        #
-        #   The +format+ variable used in this method, has to be given
-        #   as options by the user via the +options+ hash input.
-        #
-        #   Signed off by @athityakumar on 03/06/2017 at 7:00PM
         def call
-          book  = Spreadsheet::Workbook.new
-          sheet = book.create_worksheet
+          @book  = Spreadsheet::Workbook.new
+          @sheet = @book.create_worksheet
+
+          process_offsets
+          write_headers
+
+          @dataframe.each_row_with_index.with_index do |(row, idx), i|
+            write_index(idx, i+@row_offset)
+            write_data(row,  i+@row_offset)
+          end
+
+          @book.write(@path)
+        end
+
+        private
+
+        def process_offsets
+          @row_offset   = @header ? 1 : 0
+          @col_offset   = 0 unless @index
+          @col_offset ||= @dataframe.index.is_a?(Daru::MultiIndex) ? @dataframe.index.levels.size : 1
+        end
+
+        def write_headers
+          return unless @header
+
+          @sheet.row(0).concat([' '] * @col_offset + @dataframe.vectors.map(&:to_s))
+          return unless @header.is_a?(Hash)
+
+          @sheet.row(0).default_format = Spreadsheet::Format.new(@header)
+        end
+
+        def write_index(idx, row)
+          return unless @index
+
+          @sheet.row(row).concat(idx.is_a?(Array) ? idx.to_a.map(&:to_s) : [idx.to_s])
+          return unless @index.is_a?(Hash)
+
+          @col_offset.times { |col| @sheet.row(row).set_format(col, Spreadsheet::Format.new(@index)) }
```

Comment:
For a bit cleaner code, I'd define helper method:
```ruby
def format(col_range, row, format)
  return unless format.is_a?(Hash)
  ....
end
```
Then here you can just
```ruby
format(0...@col_offset, row, @index)
```
and below
```ruby
format(@col_offset...@col_offset + @dataframe.ncols, idx, @data)
```
WDYT?..